### PR TITLE
[kjobctl] Describe only kjobctl owned objects.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Describe only kjobctl owned objects on describe command.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Unfortunately, filtering by labels isn't possible when searching for resources by name. However, we can apply filters on the client side.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by IrvingMg** [comment]
LGTM.

File: cmd/experimental/kjobctl/pkg/cmd/describe/describe.go [hunk 1]
```diff
@@ -191,6 +191,11 @@ func (o *DescribeOptions) Run(ctx context.Context) error {
 			continue
 		}
 
+		labels := obj.GetLabels()
```
Comment (trasc) [inline cmd/experimental/kjobctl/pkg/cmd/describe/describe.go:194]: can't we solve this here 

https://github.com/kubernetes-sigs/kueue/blob/ba37646431e0e3eedfad874a09692df7dc32a0e2/cmd/experimental/kjobctl/pkg/cmd/describe/describe.go#L245-L260

and avoid listing unnecessary objects?

Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/pkg/cmd/describe/describe.go:194]: Unfortunately no. ResourceBuilder does not allow to filter by label selectors and name.

```
Error: name cannot be provided when a selector is specified
```

**Review by trasc — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**